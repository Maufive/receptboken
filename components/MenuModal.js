import React, { Component } from "react";
import PropTypes from "prop-types";
import Modal from "react-modal";
import Link from "next/link";
import Router from "next/router";
import UsersBoldIcon from "../svg/usersBold.svg";
import HeartBoldIcon from "../svg/likeBold.svg";
import ListBoldIcon from "../svg/listBold.svg";
import LogoutIcon from "../svg/logout.svg";
import MenuIcon from "../svg/menu-4.svg";
import CloseIcon from "../svg/closeBold.svg";
import AvatarIcon from "../svg/defaultAvatar.svg";
import {
	UserItem,
	Option,
	MenuContainer,
	ImgStyles
} from "./styles/DropdownStyles";
import { LoginContainer } from "./styles/Jumbotron";

Modal.setAppElement("#__next");

const ModalStyles = {
	content: {
		top: "50%",
		left: "50%",
		right: "auto",
		bottom: "auto",
		marginRight: "-50%",
		transform: "translate(-50%, -50%)",
		background: "rgba(21, 189, 118, 0.8)",
		width: "100vw",
		height: "100vh",
		zIndex: 10
	}
};

class UserDropdown extends Component {
	state = {
		open: false,
		isModalOpen: false
	};

	openModal = () => {
		this.setState({ isModalOpen: true });
	};

	closeModal = () => {
		this.setState({ isModalOpen: false });
	};

	logout = e => {
		localStorage.removeItem("jwtToken");
		Router.push("/");
		location.reload();
	};

	render() {
		const { user } = this.props;
		return (
			<LoginContainer>
				<UserItem onClick={this.openModal}>
					<MenuIcon />
					{user.fname}
					<ImgStyles>
						{user.photo ? (
							<img src={user.photo} alt="profilbild" />
						) : (
							<AvatarIcon />
						)}
					</ImgStyles>
				</UserItem>
				<Modal
					isOpen={this.state.isModalOpen}
					onAfterOpen={this.afterOpenModal}
					onRequestClose={this.closeModal}
					contentLabel="Menu Modal"
					style={ModalStyles}
					closeTimeoutMS={800}
				>
					<MenuContainer>
						<div>
							<Option onClick={this.closeModal}>
								<a>
									<CloseIcon />
									Stäng
								</a>
							</Option>
						</div>
						<div>
							<Option onClick={this.closeModal}>
								<Link
									href={{
										pathname: "/profile",
										query: {
											id: user._id
										}
									}}
								>
									<a>
										<UsersBoldIcon />
										Min profil
									</a>
								</Link>
							</Option>
						</div>
						<div>
							<Option onClick={this.closeModal}>
								<Link href="/saved">
									<a>
										<HeartBoldIcon />
										Sparade Recept
									</a>
								</Link>
							</Option>
						</div>
						<div>
							<Option onClick={this.closeModal}>
								<Link
									href={{
										pathname: "/inkopslistor",
										query: {
											id: user._id
										}
									}}
								>
									<a>
										<ListBoldIcon />
										Inköpslistor
									</a>
								</Link>
							</Option>
						</div>
						<div>
							<Option onClick={this.logout}>
								<a>
									<LogoutIcon />
									Logga ut
								</a>
							</Option>
						</div>
					</MenuContainer>
				</Modal>
			</LoginContainer>
		);
	}
}

export default UserDropdown;

UserDropdown.propTypes = {
	user: PropTypes.object.isRequired
};
