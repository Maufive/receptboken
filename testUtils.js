export const fakeRecept = {
	_id: "1234",
	author: "5678436374re67fwuf76",
	created: "Idag",
	description: ["Steg1", "Steg2", "Steg3"],
	id: "1337",
	ingredients: [
		{
			input: "Lök",
			numberOfUnits: "2",
			units: "dl"
		}
	],
	photo: "asdf123.jpg",
	reviews: [{ score: 1 }],
	servings: "2",
	tags: ["Mat", "Gott"],
	timeRequired: "20 min",
	title: "Mitt jättegoda recept"
};

export const fakeUser = {
	_id: "1234",
	username: "fakeuser@gmail.com",
	password: "1234",
	hearts: [],
	fname: "Niklas",
	lname: "Albinsson",
	description: "Hej jag heter Niklas",
	photo: "http://fakephoto.jpg"
};

export const fakeShoppinglist = [
	{
		_id: "34567876543452176",
		title: "My Shoppinglist",
		created: "12/3 2019",
		author: "61386712376576d7q6ewd65",
		list: [
			{
				input: "Lök",
				numberOfUnits: "5",
				units: "dl"
			},
			{
				input: "Tomater",
				numberOfUnits: "10",
				units: "st"
			}
		]
	}
];
