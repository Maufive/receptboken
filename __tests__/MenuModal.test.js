import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import MenuModal from "../components/MenuModal";
import { fakeUser } from "../testUtils";

describe("<MenuModal />", () => {
	let wrapper;
	beforeEach(() => (wrapper = shallow(<MenuModal user={fakeUser} />)));

	it("renders with closed state", () => {
		expect(wrapper.find("Modal").prop("isOpen")).toEqual(false);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	it("renders with open state", () => {
		wrapper.setState({ isModalOpen: true });
		expect(wrapper.find("Modal").prop("isOpen")).toEqual(true);
	});

	it("opens modal when button is clicked", () => {
		const wrapper = shallow(<MenuModal user={fakeUser} />);
		wrapper.find("DropdownStyles__UserItem").simulate("click");
		wrapper.update();
		expect(wrapper.state("isModalOpen")).toBe(true);
	});
});
