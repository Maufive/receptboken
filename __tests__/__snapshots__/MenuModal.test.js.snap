// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuModal /> renders with closed state 1`] = `
<Jumbotron__LoginContainer>
  <DropdownStyles__UserItem
    onClick={[Function]}
  >
    <MenuIcon
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    />
    Niklas
    <DropdownStyles__ImgStyles>
      <img
        alt="profilbild"
        src="http://fakephoto.jpg"
      />
    </DropdownStyles__ImgStyles>
  </DropdownStyles__UserItem>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={800}
    contentLabel="Menu Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "background": "rgba(78, 183, 49, 0.9)",
          "bottom": "auto",
          "height": "100vh",
          "left": "50%",
          "marginRight": "-50%",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "100vw",
          "zIndex": 10,
        },
      }
    }
  >
    <DropdownStyles__MenuContainer>
      <div>
        <DropdownStyles__Option
          onClick={[Function]}
        >
          <a>
            <CloseIcon
              viewBox="0 0 32.526 32.526"
              xmlns="http://www.w3.org/2000/svg"
            />
            Stäng
          </a>
        </DropdownStyles__Option>
      </div>
      <div>
        <DropdownStyles__Option
          onClick={[Function]}
        >
          <Link
            href={
              Object {
                "pathname": "/profile",
                "query": Object {
                  "id": "1234",
                },
              }
            }
          >
            <a>
              <UsersBoldIcon
                viewBox="0 0 60 60"
                xmlns="http://www.w3.org/2000/svg"
              />
              Min profil
            </a>
          </Link>
        </DropdownStyles__Option>
      </div>
      <div>
        <DropdownStyles__Option
          onClick={[Function]}
        >
          <Link
            href="/saved"
          >
            <a>
              <HeartBoldIcon
                viewBox="0 0 51.997 51.997"
                xmlns="http://www.w3.org/2000/svg"
              />
              Sparade Recept
            </a>
          </Link>
        </DropdownStyles__Option>
      </div>
      <div>
        <DropdownStyles__Option
          onClick={[Function]}
        >
          <Link
            href={
              Object {
                "pathname": "/inkopslistor",
                "query": Object {
                  "id": "1234",
                },
              }
            }
          >
            <a>
              <ListBoldIcon
                viewBox="0 0 60 60"
                xmlns="http://www.w3.org/2000/svg"
              />
              Inköpslistor
            </a>
          </Link>
        </DropdownStyles__Option>
      </div>
      <div>
        <DropdownStyles__Option
          onClick={[Function]}
        >
          <a>
            <LogoutIcon
              viewBox="0 0 129 129"
              xmlns="http://www.w3.org/2000/svg"
            />
            Logga ut
          </a>
        </DropdownStyles__Option>
      </div>
    </DropdownStyles__MenuContainer>
  </Modal>
</Jumbotron__LoginContainer>
`;
